src/Cache/Engine/ApcEngine.php

```diff
@@ -149,14 +169,20 @@ public function clear($check)
                 '/^' . preg_quote($this->_config['prefix'], '/') . '/',
                 APC_ITER_NONE
             );
-            apcu_delete($iterator);
 
-            return true;
+            return apcu_delete($iterator) === true;
```

Comment:
This should throw an exception, too, instead of returning false?
Reason being: the return type for clear was always `bool true` and now it can suddenly be `bool false` and people would probably not check if clearing was successful. An exception would tell them that it wasn't?

Reply:
We don't throw exceptions from other cache engines when deletes fail though.

Reply:
And that's good behavior?

Reply:
We could also make this configurable for now and switch around the defaults to 'throw exception' in 4.x?

Reply:
Generally we don't want caching operations to destroy an inflight request. Caching should be used as an optimization or performance improvement not as critical path code that needs to be 100% reliable. At least that is how I approach cache usage.

Reply:
Should we throw a warning then, instead? Maybe for all the exceptions I was adding here?
Make the throwing of the warning configurable (true by default)?

Reply:
Using trigger_error() might be a good option here, as that will generally end up in the logs.